lib/rules/media-feature-name-unit-allowed-list/index.js

```diff
@@ -0,0 +1,99 @@
+'use strict';
+
+const report = require('../../utils/report');
+const ruleMessages = require('../../utils/ruleMessages');
+const validateOptions = require('../../utils/validateOptions');
+const { TokenType } = require('@csstools/css-tokenizer');
+const { isTokenNode } = require('@csstools/css-parser-algorithms');
+const {
+	isMediaFeaturePlain,
+	isMediaFeatureRange,
+	parse,
+} = require('@csstools/media-query-list-parser');
+const validateObjectWithArrayProps = require('../../utils/validateObjectWithArrayProps');
+const { isString } = require('../../utils/validateTypes');
+const matchesStringOrRegExp = require('../../utils/matchesStringOrRegExp');
+const flattenArray = require('../../utils/flattenArray');
+const atRuleParamIndex = require('../../utils/atRuleParamIndex');
+
+const ruleName = 'media-feature-name-unit-allowed-list';
+
+const messages = ruleMessages(ruleName, {
+	rejected: (unit, name) => `Unexpected unit "${unit}" for name "${name}"`,
+});
+
+const meta = {
+	url: 'https://stylelint.io/user-guide/rules/media-feature-name-unit-allowed-list',
+};
+
+/** @type {import('stylelint').Rule} */
+const rule = (primary) => {
+	return (root, result) => {
+		const validOptions = validateOptions(result, ruleName, {
+			actual: primary,
+			possible: [validateObjectWithArrayProps(isString)],
+		});
+
+		if (!validOptions) {
+			return;
+		}
+
+		root.walkAtRules(/^media$/i, (atRule) => {
+			const mediaQueryList = parse(atRule.params);
+
+			mediaQueryList.forEach((mediaQuery) => {
+				mediaQuery.walk((entry) => {
+					if (!isMediaFeaturePlain(entry.node) && !isMediaFeatureRange(entry.node)) {
+						return;
+					}
+
+					const name = isMediaFeatureRange(entry.node)
+						? entry.node.name.name.value[1].toString()
+						: entry.node.name.toString();
```

Comment:
This differs from @romainmenke's suggested code since it didn't properly handle range contexts for Level 4 Mediq Queries. I only figured out this "hardcoded" position from inspecting the generated tree.

Should I:

- document this somewhere outside of this PR
- have a more robust way of doing this

Or, am I missing a feature of the library?

Reply:
This will become simpler in a future version.
Most likely : `entry.node.getName()`

-----

Can you try `entry.node.name.getName()`?
This doesn't yet work for boolean media query features but we are only handling range and plain features here so the `getName` function exists for both.

`getName()` also ensures that you are using unescaped values.

```css
@media (\61 spect-ratio: 1/1) {}
@media (aspect-ratio: 1/1) {}
```

These are exactly the same but `name.toString()` will have the raw (escaped) form and `name.getName()` will have the unescaped form.

Reply:
Ah, great suggestion - didn't realize this function existed. Just updated!